NOTES.md

```diff
@@ -1,3 +1,186 @@
+re-import the affected tables using the `rethinkdb dump` and
+`rethinkdb restore` commands. See [Back up your data][backup-docs] for
+more information.
+
+If you are unsure if any of your documents are affected, you can run
+`python -m rethinkdb._negative_zero_check` after upgrading both the
+server and Python driver.  See the output of `python -m
+rethinkdb._negative_zero_check --help` for additional options.
+
+[1.14-outdated-index]: http://rethinkdb.com/docs/troubleshooting/#my-secondary-index-is-outdated
+[backup-docs]: http://www.rethinkdb.com/docs/backup/
+
+### API-breaking changes ###
+
+* `between` no longer accepts `null` bounds. The new `r.minval` and
+  `r.maxval` can be used instead
+
+* The `any` and `all` commands have been removed. The `or` and `and`
+  commands can be used instead
+
+* `indexes_of` has been renamed to `offsets_of`
+
+* The `squash` argument to `changes` now defaults to `false`
+
+* The type hierarchy for exception types in the Python driver changed.
+  All exceptions including `RqlDriverError` now inherit from the
+  `RqlError` type.  `RqlRuntimeError`, `RqlCompileError` and
+  `RqlClientError` additionally inherit from the new `RqlQueryError`
+  type.
+
+## New Features ##
+
+* ReQL
+  * Added support for changefeeds on `get_all` and `union` queries (#3642)
+  * `between` no longer accepts `null` as a bound. The new `r.minval` and `r.maxval` can be used instead (#1023)
+  * Added support for getting the state of a changefeed using the new `include_states` optarg to `changes` (#3709)
+* Drivers
+  * Added support for non-blocking `cursor.get` (#3529)
+  * Added support for parallel query execution on a single connection (#3754)
+  * Consolidated the return types and use the new `ResponseNotes` field to convey extra information (#3715)
+* Python driver
+  * Add an optional script that warns for documents with negative zero as a primary key (#3637)
+
+## Improvements ##
+
+* Server
+  * Report open cursors as a single entry in the jobs table (#3662)
+  * No longer send timestamps between servers in `batchspec_t` (#2671)
+  * No longer perform some expensive checks in release mode (#3656)
+  * Include the remote port number in the heartbeat timeout message (#2891)
+  * Improved the ordering and throttling of reads and writes (#1606)
+  * Limit the default number of documents per batch to reduce latency (#3806)
+  * Execute multiple queries in parallel on a single connection (#3296)
+  * Improved the performance of sending responses (#3744)
+* Web UI
+  * Added a modifiable query limit (#3910)
+  * Add an "add table" button to each database (#3522)
+* ReQL
+  * `table.rebalance` with insufficient data is no longer an error (#3679)
+  * Renamed `indexes_of` to `offsets_of` to avoid confusion with secondary indexes (#3265)
+  * Removed `any` and `all` in favor of `or` and `and` (#1581)
+  * Immediately return an empty first batch when the result is a changefeed (#3852)
```

Comment:
I would put this under "server" and write "send back" instead of "return".

The empty batch is completely invisible to the user.